flexget/plugins/search_kat.py

```diff
@@ -69,34 +78,66 @@ def search(self, task, entry, config):
                 elif r.status_code != 200:
                     log.warning('Search returned %s response code' % r.status_code)
                     continue
-                rss = feedparser.parse(r.content)
 
-                ex = rss.get('bozo_exception', False)
-                if ex:
-                    log.warning('Got bozo_exception (bad feed)')
-                    continue
+                entries |= self.parse_rss(r.content) if config.get('rss') else self.parse_web(r.content)
 
-                for item in rss.entries:
-                    entry = Entry()
-                    entry['title'] = item.title
+        return entries
+
+    def parse_rss(self, content):
+        log.debug("using rss feed")
+        entries = set()
 
-                    if not item.get('enclosures'):
-                        log.warning('Could not get url for entry from KAT. Maybe plugin needs updated?')
-                        continue
-                    entry['url'] = item.enclosures[0]['url']
-                    entry['torrent_seeds'] = int(item.torrent_seeds)
-                    entry['torrent_leeches'] = int(item.torrent_peers)
-                    entry['search_sort'] = torrent_availability(entry['torrent_seeds'], entry['torrent_leeches'])
-                    entry['content_size'] = int(item.torrent_contentlength) / 1024 / 1024
-                    entry['torrent_info_hash'] = item.torrent_infohash
+        rss = feedparser.parse(content)
+        ex = rss.get('bozo_exception', False)
+        if ex:
+            log.warning('Got bozo_exception (bad feed)')
+            return entries
 
-                    entries.add(entry)
+        for item in rss.entries:
+            entry = Entry()
+            entry['title'] = item.title
 
-                if len(rss.entries) < 25:
-                    break
+            if not item.get('enclosures'):
+                log.warning('Could not get url for entry from KAT. Maybe plugin needs updated?')
+                continue
+            entry['url'] = item.enclosures[0]['url']
+            entry['torrent_seeds'] = int(item.torrent_seeds)
+            entry['torrent_leeches'] = int(item.torrent_peers)
+            entry['search_sort'] = torrent_availability(entry['torrent_seeds'], entry['torrent_leeches'])
+            entry['content_size'] = int(item.torrent_contentlength) / 1024 / 1024
+            entry['torrent_info_hash'] = item.torrent_infohash
+
+            entries.add(entry)
 
         return entries
 
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Method could be a function](https://www.codacy.com/app/kristen/Flexget/file/2167275111/issues/source?bid=3161667&fileBranchId=3161675#l114)